### ISSUE TITLE
Panfrost OpenGL2/3 render method for Mali graphics 

### ISSUE THREAD

**Issue by Llerey** [FCL-Team/FoldCraftLauncher#370]
### Description

Found documentation for renderer support for some Mali graphics accelerators

### Why

The Panfrost driver stack includes an OpenGL ES implementation for Arm Mali GPUs based on the Midgard and Bifrost microarchitectures. It is conformant on Mali-G52 and Mali-G57. 

More details in the article: more details in the documentation:
https://docs.mesa3d.org/drivers/panfrost.html

**Comment by Llerey** [comment]
https://github.com/Saikatsaha1996/mesa-Panfrost-G610
This is related to the article above

**Comment by Llerey** [comment]
I checked the information about this, these architectures are included in the main Mesa driver, it needs to be enabled at compile time, and use the panfrost compiler.

**Comment by FrostNovaHD** [comment]
> I checked the information about this, these architectures are included in the main Mesa driver, it needs to be enabled at compile time, and use the panfrost compiler.

Yes do it needs a kernel module that isn't on your android devices

**Comment by Llerey** [comment]
> > Я проверил информацию об этом, эти архитектуры включены в основной драйвер Mesa, его нужно включить во время компиляции, и использовать компилятор panfrost.
> 
> Да, нужен модуль ядра, которого нет на ваших устройствах Android.

I don’t quite understand how it works, but the general principle seems to be the same as that of zink or virgle 
https://github.com/Saikatsaha1996/mesa-Panfrost-G610

**Comment by Llerey** [comment]
It would be interesting to try using some visualizer that is already supported by the launcher with this flag

**Comment by FrostNovaHD** [comment]
> .
> 
> I don’t quite understand how it works, but the general principle seems to be the same as that of zink or virgle https://github.com/Saikatsaha1996/mesa-Panfrost-G610

The thing you posted only works on Mali g610 and g710 pojav tried it

**Comment by Llerey** [comment]
I'll try to run panfrost with the configuration for Valhalla Mali G57 in Termux

**Comment by Llerey** [comment]
I will write you later

**Comment by Llerey** [comment]
> > .
> > I don’t quite understand how it works, but the general principle seems to be the same as that of zink or virgle https://github.com/Saikatsaha1996/mesa-Panfrost-G610
> 
> The thing you posted only works on Mali g610 and g710 pojav tried it

How was the performance?

**Comment by FrostNovaHD** [comment]
> > > .
> > > I don’t quite understand how it works, but the general principle seems to be the same as that of zink or virgle https://github.com/Saikatsaha1996/mesa-Panfrost-G610
> > 
> > 
> > The thing you posted only works on Mali g610 and g710 pojav tried it
> 
> How was the performance?

Poor

**Comment by Mateus2022** [comment]
Hello for all,

I'm an old driver developer (more curious in one part) about panfrost that modification of the g610 for me is functional in the Bifrost branch, but as I tested it in the proot environment the fps is between 0-1 and there is no image in x11

that is an old modification of icecream95 which has a lot of bugs and is not functional on several gpus, what we should do is wait for panfrost to receive standard support from the mali gpu kernel before it can leave the mesa.

**Comment by joaobatista070** [comment]
(for original panfrost mesa drm panfrost/panthor is need on kernel) but panfork mesa or mesa panfrost g610 is bad modificativo to use default Mali kernel

**Comment by JhosephRF** [comment]
My Mali-G68 gpu is compatible with the panfrost from mesa 3d, I tried to use it in termux but it requires a kernel with the panfrost module "config_drm_panfrost". I did some Googling and this was the most I could find about to panfrost:

(An old tutorial)

https://www.kernelconfig.io/config_drm_panfrost

(Driver for Box86/64 and Exagear)

https://github.com/Matews371/Panfrost-Android
